conformance/tests/httproute-timeouts.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteTimeoutRequest)
+	ConformanceTests = append(ConformanceTests, HTTPRouteTimeoutBackendRequest)
+}
+
+// Note that here we are only making sure Request and BackendRequest timeouts work individually.
+// TODO: once we add retry support to Gateway API (future GEP-1731), add conformance tests to show
+// Request and BackendRequest timeouts working together.
+
+var HTTPRouteTimeoutRequest = suite.ConformanceTest{
+	ShortName:   "HTTPRouteTimeoutRequest",
+	Description: "An HTTPRoute with request timeout",
+	Manifests:   []string{"tests/httproute-timeout-request.yaml"},
+	Features: []suite.SupportedFeature{
+		suite.SupportGateway,
+		suite.SupportHTTPRoute,
+		suite.SupportHTTPRouteRequestTimeout,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ns := "gateway-conformance-infra"
+		routeNN := types.NamespacedName{Name: "request-timeout", Namespace: ns}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request:  http.Request{Path: "/"},
+				Response: http.Response{StatusCode: 200},
+			}, {
+				Request:  http.Request{Path: "/?delay=1s"},
+				Response: http.Response{StatusCode: 504},
+			},
+		}
+
+		for i := range testCases {
+			// Declare tc here to avoid loop variable
+			// reuse issues across parallel tests.
+			tc := testCases[i]
+			t.Run(tc.GetTestCaseName(i), func(t *testing.T) {
+				t.Parallel()
+				http.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, tc)
+			})
+		}
+	},
+}
+
+var HTTPRouteTimeoutBackendRequest = suite.ConformanceTest{
```

Comment:
Nit: Prefer to have test file names directly match test names + manifest names, so would rather split this out to a separate file.